core/src/main/scala/kafka/server/AlterPartitionManager.scala

```diff
@@ -270,7 +270,7 @@ class DefaultAlterPartitionManager(
 
     inflightAlterPartitionItems.groupBy(_.topicIdPartition.topic).foreach { case (topicName, items) =>
       val topicId = items.head.topicIdPartition.topicId
-      canUseTopicIds &= topicId != Uuid.ZERO_UUID
+      canUseTopicIds &= !topicId.equals(Uuid.ZERO_UUID)
```

Comment:
IIRC, it calls the equals method on the first object in Scala.

Reply:
right, we don't need to do this for Scala files, reverted change in https://github.com/apache/kafka/pull/15707/commits/e206a2cfdfa955b3ab004b5644e28e6fa23999b9